spec.html

```diff
@@ -4906,6 +4980,89 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
+        1. For each element _entry_ in _O_.[[PrivateFieldValues]],
+          1. If _entry_.[[PrivateName]] is _P_, return _entry_.
+        1. Return ~empty~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldadd" aoid="PrivateFieldAdd">
+      <h1>PrivateFieldAdd ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
+        1. Append { [[PrivateName]]: P, [[PrivateFieldValue]]: _value_ } to _O_.[[PrivateFieldValues]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldget" aoid="PrivateFieldGet">
+      <h1>PrivateFieldGet ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is ~empty~, throw a *TypeError* exception.
+        1. Return _entry_.[[PrivateFieldValue]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldset" aoid="PrivateFieldSet">
+      <h1>PrivateFieldSet ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is ~empty~, throw a *TypeError* exception.
+        1. Set _entry_.[[PrivateFieldValue]] to _value_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-definefield" aoid="DefineField">
+      <h1>DefineField ( _receiver_, _fieldRecord_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_receiver_) is Object.
+        1. Assert: _fieldRecord_ is a Record as created by ClassFieldDefinitionEvaluation.
+        1. Let _name_ be _fieldRecord_.[[Name]].
+        1. Let _initializer_ be _fieldRecord_.[[Initializer]].
+        1. If _initializer_ is not ~empty~, then
+          1. Let _initValue_ be ? Call(_initializer_, _receiver_).
+        1. Else, let _initValue_ be *undefined*.
+        1. If _fieldRecord_.[[IsAnonymousFunctionDefinition]] is *true*, then
+          1. Let _hasNameProperty_ be ? HasOwnProperty(_initValue_, `"name"`).
+          1. If _hasNameProperty_ is *false*, perform SetFunctionName(_initValue_, _fieldName_).
+        1. If _fieldName_ is a Private Name,
+          1. Perform ? PrivateFieldAdd(_fieldName_, _receiver_, _initValue_).
+        1. Else,
+          1. Assert: IsPropertyKey(_fieldName_) is *true*.
+          1. Perform ? CreateDataPropertyOrThrow(_receiver_, _fieldName_, _initValue_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-initializepublicinstancefields" aoid="InitializeInstanceFields">
+      <h1>InitializeInstanceFields ( _O_, _constructor_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_O_) is Object.
+        1. Assert: _constructor_ is an ECMAScript function object.
```

Comment:
would it be cleaner to do this?
```suggestion
        1. Assert: ! IsCallable(_constructor_) is *true*.
```

Reply:
Cleaner, maybe, but doesn't mean the same. `MakeConstructor` already has the proposed wording.